bot_interface.py

```diff
@@ -0,0 +1,304 @@
+#       }
+# <user_id_2> : ...
+users_data = {}
+messages_to_delete = {}
+
+if os.path.exists(backup_messages_filename):
+    with open(backup_messages_filename, 'rb') as f:
+        messages_to_delete = pickle.load(f)
+        for user_id, messages_ids in messages_to_delete.items():
+            if messages_ids:
+                warning_message = bot.send_message(user_id, 'Сервер был перезапущен. Нажмите /start, чтобы начать заново.')
+                users_data[user_id] = {}
+                users_data[user_id]['messages_ids'] = {}
+                users_data[user_id]['messages_ids']['warning_messages'] = warning_message.message_id
+                for message_id in messages_ids:
+                    try:
+                        bot.delete_message(user_id, message_id)
+                    except Exception as e:
+                        if 'message to delete not found' in str(e):
+                            pass
+                        else:
+                            raise
+                messages_to_delete[user_id] = list()
+                messages_to_delete[user_id].append(warning_message.message_id)
+
+
+def dump_data_to_file(messages):
+    with open(backup_messages_filename, 'wb') as f:
+        pickle.dump(messages, f)
+
+
+def check_for_warning(user_id):
+    if 'messages_ids' in users_data[user_id] and 'warning_message' in users_data[user_id]['messages_ids']:
+        try:
+            bot.delete_message(user_id, users_data[user_id]['messages_ids']['warning_message'])
+        except Exception as e:
+            if 'message is not modified' in str(e):
+                pass
+            else:
+                raise
+        finally:
+            users_data[user_id]['messages_ids'].pop('warning_message')
+
+
+def edit_characters_list(call, markup=None):
+    user_id = call.message.chat.id
+    current_list = '\n'.join(f"{k} - {v}" for k, v in users_data[user_id]['characters_config'].items()) \
+        if users_data[user_id]['characters_config'] else '**Пока пусто**'
+    try:
+        bot.edit_message_text(chat_id=user_id, message_id=users_data[user_id]['messages_ids']['first_message'],
+                          text=f'Список героев: \n\n{current_list}\n\n', reply_markup=markup)
+    except Exception as e:
+        if 'message is not modified' in str(e):
+            pass
+        else:
+            raise
+
+
+def handle_too_fast_click(call, user_id):
+    try:
+        current_time = time.time()
+        if 'timer' in users_data[user_id] and current_time - users_data[user_id]['timer'] < 1:
+            return bot.answer_callback_query(call.id, "Слишком быстро! Подождите немного.")
+        users_data[user_id]['timer'] = current_time
+        bot.answer_callback_query(call.id)
+    except Exception as e:
+        if 'query is too old' in str(e):
+            pass
+        else:
+            raise
+
+
+def clear_previous_messages(user_id):
+    if user_id in users_data and users_data[user_id]['messages_ids']:
```

Comment:
I guess it's enough to check only the user here: `if user_id in users_data`

Reply:
see my previous comment for line 81